We can't see the Yarplogger messages right away as they are received on the yarpjs input port

Refer to [issue 3] in the issue recap below:

> ## Test Report
> 
> ### Objectives
> The tests focused on checking the data pipe and overall functionality related to the Yarp Text Logging of the `yarprobotinterface` process running on the robot head and the module `WalkingModule` running on some other machine connected to the robot network. The related telemetry domain entries in the Open MCT Visualiser, `yarprobotinterface process logging through yarplogger ` and `WalkingModule process logging through yarplogger ` under the telemetry folder `Process Logging`:
> ![image](https://user-images.githubusercontent.com/6848872/193764201-e1b78849-c401-41fa-9d5a-00af85e8b7b6.png)
> 
> ### Test Sequence, Outcome and Issues
> - Installation went smoothly.
> - **[Issue 1]:** Running the telemetry server from the folder `./iCubTelemVizServer` and `node` command fails:
>    ```
>    $> cd iCubTelemVizServer
>    $> node iCubTelemVizServer.js
>    ...
>    /Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:282
>                this.state[id].ori.roll = sensorSample[3][0][0][0];
>                                                            ^
>    
>    TypeError: Cannot read property '0' of undefined
>        at ICubTelemetry.parseFromId (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:282:57)
>        at Object.<anonymous> (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:166:31)
>        at /Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/iCubTelemVizServer.js:74:56
>        at /Users/nunoguedelha/dev/yarp-openmct/node_modules/YarpJS/yarp.js:188:13
>    $>
>    ```
>    <details>
>    <summary>Full log</summary>
>    
>    ```
>    iCub Telemetry server launched!
>    [INFO] |yarp.os.Port|/yarpjs/inertial:i| Port /yarpjs/inertial:i active at tcp://192.168.1.70:10130/
>    [INFO] |yarp.os.Port|/yarpjs/head/inertials:i| Port /yarpjs/head/inertials:i active at tcp://192.168.1.70:10131/
>    [INFO] |yarp.os.Port|/yarpjs/leftArm/inertials:i| Port /yarpjs/leftArm/inertials:i active at tcp://192.168.1.70:10132/
>    [INFO] |yarp.os.Port|/yarpjs/rightArm/inertials:i| Port /yarpjs/rightArm/inertials:i active at tcp://192.168.1.70:10133/
>    [INFO] |yarp.os.Port|/yarpjs/leftLeg/inertials:i| Port /yarpjs/leftLeg/inertials:i active at tcp://192.168.1.70:10134/
>    [INFO] |yarp.os.Port|/yarpjs/rightLeg/inertials:i| Port /yarpjs/rightLeg/inertials:i active at tcp://192.168.1.70:10135/
>    [INFO] |yarp.os.Port|/yarpjs/leftFoot/inertials:i| Port /yarpjs/leftFoot/inertials:i active at tcp://192.168.1.70:10136/
>    [INFO] |yarp.os.Port|/yarpjs/rightFoot/inertials:i| Port /yarpjs/rightFoot/inertials:i active at tcp://192.168.1.70:10137/
>    [INFO] |yarp.os.Port|/yarpjs/left_leg/stateExt:o| Port /yarpjs/left_leg/stateExt:o active at tcp://192.168.1.70:10138/
>    [INFO] |yarp.os.Port|/yarpjs/right_leg/stateExt:o| Port /yarpjs/right_leg/stateExt:o active at tcp://192.168.1.70:10139/
>    [INFO] |yarp.os.Port|/yarpjs/left_arm/stateExt:o| Port /yarpjs/left_arm/stateExt:o active at tcp://192.168.1.70:10140/
>    [INFO] |yarp.os.Port|/yarpjs/right_arm/stateExt:o| Port /yarpjs/right_arm/stateExt:o active at tcp://192.168.1.70:10141/
>    [INFO] |yarp.os.Port|/yarpjs/torso/stateExt:o| Port /yarpjs/torso/stateExt:o active at tcp://192.168.1.70:10142/
>    [INFO] |yarp.os.Port|/yarpjs/head/stateExt:o| Port /yarpjs/head/stateExt:o active at tcp://192.168.1.70:10143/
>    [INFO] |yarp.os.Port|/yarpjs/camLeftEye:i| Port /yarpjs/camLeftEye:i active at tcp://192.168.1.70:10144/
>    [INFO] |yarp.os.Port|/yarpjs/camRightEye:i| Port /yarpjs/camRightEye:i active at tcp://192.168.1.70:10145/
>    [INFO] |yarp.os.Port|/yarpjs/left_arm/cartesianEndEffectorWrench:i| Port /yarpjs/left_arm/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10146/
>    [INFO] |yarp.os.Port|/yarpjs/right_arm/cartesianEndEffectorWrench:i| Port /yarpjs/right_arm/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10147/
>    [INFO] |yarp.os.Port|/yarpjs/left_upper_leg/cartesianEndEffectorWrench:i| Port /yarpjs/left_upper_leg/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10148/
>    [INFO] |yarp.os.Port|/yarpjs/left_lower_leg/cartesianEndEffectorWrench:i| Port /yarpjs/left_lower_leg/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10149/
>    [INFO] |yarp.os.Port|/yarpjs/right_upper_leg/cartesianEndEffectorWrench:i| Port /yarpjs/right_upper_leg/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10150/
>    [INFO] |yarp.os.Port|/yarpjs/right_lower_leg/cartesianEndEffectorWrench:i| Port /yarpjs/right_lower_leg/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10151/
>    [INFO] |yarp.os.Port|/yarpjs/left_foot_front/cartesianEndEffectorWrench:i| Port /yarpjs/left_foot_front/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10152/
>    [INFO] |yarp.os.Port|/yarpjs/left_foot_rear/cartesianEndEffectorWrench:i| Port /yarpjs/left_foot_rear/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10153/
>    [INFO] |yarp.os.Port|/yarpjs/right_foot_front/cartesianEndEffectorWrench:i| Port /yarpjs/right_foot_front/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10154/
>    [INFO] |yarp.os.Port|/yarpjs/right_foot_rear/cartesianEndEffectorWrench:i| Port /yarpjs/right_foot_rear/cartesianEndEffectorWrench:i active at tcp://192.168.1.70:10155/
>    [INFO] |yarp.os.Port|/yarpjs/left_arm/FT:i| Port /yarpjs/left_arm/FT:i active at tcp://192.168.1.70:10156/
>    [INFO] |yarp.os.Port|/yarpjs/right_arm/FT:i| Port /yarpjs/right_arm/FT:i active at tcp://192.168.1.70:10157/
>    [INFO] |yarp.os.Port|/yarpjs/left_leg_hip/FT:i| Port /yarpjs/left_leg_hip/FT:i active at tcp://192.168.1.70:10158/
>    [INFO] |yarp.os.Port|/yarpjs/right_leg_hip/FT:i| Port /yarpjs/right_leg_hip/FT:i active at tcp://192.168.1.70:10159/
>    [INFO] |yarp.os.Port|/yarpjs/left_foot_heel_tiptoe/FT:i| Port /yarpjs/left_foot_heel_tiptoe/FT:i active at tcp://192.168.1.70:10160/
>    [INFO] |yarp.os.Port|/yarpjs/right_foot_heel_tiptoe/FT:i| Port /yarpjs/right_foot_heel_tiptoe/FT:i active at tcp://192.168.1.70:10161/
>    [INFO] |yarp.os.Port|/yarpjs/battery/data:i| Port /yarpjs/battery/data:i active at tcp://192.168.1.70:10162/
>    [INFO] |yarp.os.Port|/yarpjs/walking-coordinator-logger/data:i| Port /yarpjs/walking-coordinator-logger/data:i active at tcp://192.168.1.70:10163/
>    [INFO] |yarp.os.Port|/yarpjs/proc-yarprobotinterface-yarptextlogging:i| Port /yarpjs/proc-yarprobotinterface-yarptextlogging:i active at tcp://192.168.1.70:10164/
>    [INFO] |yarp.os.Port|/yarpjs/proc-walkingmodule-yarptextlogging:i| Port /yarpjs/proc-walkingmodule-yarptextlogging:i active at tcp://192.168.1.70:10165/
>    [INFO] |yarp.os.Port|/yarpjs/sysCmdsGenerator/rpc| Port /yarpjs/sysCmdsGenerator/rpc active at tcp://192.168.1.70:10166/
>    { status: 'OK', message: 'Opem-MCT static server process started.' }
>    info:Yarp command successfully started...
>    ICubTelemetry History Server listening on http://192.168.1.70:8081/history
>    ICubTelemetry Realtime Server listening on ws://192.168.1.70:8081/realtime
>    Control Console Server listening on http://192.168.1.70:3000
>    Yarp port names retrieval completed successfully (0)
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/inertial:i| Receiving input from /icubSim/inertial to /yarpjs/inertial:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/head/inertials:i| Receiving input from /icubSim/head/inertials/measures:o to /yarpjs/head/inertials:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/left_leg/stateExt:o| Receiving input from /icubSim/left_leg/stateExt:o to /yarpjs/left_leg/stateExt:o using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/right_leg/stateExt:o| Receiving input from /icubSim/right_leg/stateExt:o to /yarpjs/right_leg/stateExt:o using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/left_arm/stateExt:o| Receiving input from /icubSim/left_arm/stateExt:o to /yarpjs/left_arm/stateExt:o using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/right_arm/stateExt:o| Receiving input from /icubSim/right_arm/stateExt:o to /yarpjs/right_arm/stateExt:o using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/torso/stateExt:o| Receiving input from /icubSim/torso/stateExt:o to /yarpjs/torso/stateExt:o using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/head/stateExt:o| Receiving input from /icubSim/head/stateExt:o to /yarpjs/head/stateExt:o using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/left_arm/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/left_arm/cartesianEndEffectorWrench:o to    /yarpjs/left_arm/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/right_arm/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/right_arm/cartesianEndEffectorWrench:o to    /yarpjs/right_arm/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/left_upper_leg/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/left_upper_leg/cartesianEndEffectorWrench:o to    /yarpjs/left_upper_leg/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/left_lower_leg/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/left_lower_leg/cartesianEndEffectorWrench:o to    /yarpjs/left_lower_leg/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/right_upper_leg/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/right_upper_leg/cartesianEndEffectorWrench:o to    /yarpjs/right_upper_leg/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/right_lower_leg/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/right_lower_leg/cartesianEndEffectorWrench:o to    /yarpjs/right_lower_leg/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/left_foot_front/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/left_foot_front/cartesianEndEffectorWrench:o to    /yarpjs/left_foot_front/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/left_foot_rear/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/left_foot_rear/cartesianEndEffectorWrench:o to    /yarpjs/left_foot_rear/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/right_foot_front/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/right_foot_front/cartesianEndEffectorWrench:o to    /yarpjs/right_foot_front/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/right_foot_rear/cartesianEndEffectorWrench:i| Receiving input from /wholeBodyDynamics/right_foot_rear/cartesianEndEffectorWrench:o to    /yarpjs/right_foot_rear/cartesianEndEffectorWrench:i using tcp
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/proc-yarprobotinterface-yarptextlogging:i| Receiving input from /log/iiticublap199/yarprobotinterfa[yarprobotinterface]/85559 to    /yarpjs/proc-yarprobotinterface-yarptextlogging:i using tcp
>    [OPEN-MCT STATIC SERVER] ipc: {"pid":85881}
>    [INFO] |yarp.os.impl.PortCoreInputUnit|/yarpjs/proc-walkingmodule-yarptextlogging:i| Receiving input from /log/iiticublap199/WalkingModule-0.[walkingModule]/85560 to /yarpjs/proc-   walkingmodule-yarptextlogging:i using tcp
>    /Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:282
>                this.state[id].ori.roll = sensorSample[3][0][0][0];
>                                                            ^
>    
>    TypeError: Cannot read property '0' of undefined
>        at ICubTelemetry.parseFromId (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:282:57)
>        at Object.<anonymous> (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:166:31)
>        at /Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/iCubTelemVizServer.js:74:56
>        at /Users/nunoguedelha/dev/yarp-openmct/node_modules/YarpJS/yarp.js:188:13
>    ```
>    
>    </details>
> - The app could be launched without issues from the repo root:
>    ```
>    npm start
>    ```
> - **[Issue 2]:** As soon as the server starts, it crashes with the following issue:
>    ```
>    /Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:404
>        this.history[id].push(telemetrySample); // update history
>                         ^
>    
>    TypeError: Cannot read property 'push' of undefined
>        at ICubTelemetry.generateTelemetry (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:404:22)
>        at ICubTelemetry.<anonymous> (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:183:18)
>        at Array.forEach (<anonymous>)
>        at ICubTelemetry.<anonymous> (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:182:33)
>        at listOnTimeout (internal/timers.js:555:17)
>        at processTimers (internal/timers.js:498:7)
>    ```
>    
>    <details>
>    <summary>Extended log</summary>
> 
>    ```
>    NEW CONNECTION!!
>    CLOSE CONNECTION
>    [OPEN-MCT STATIC SERVER] stdout: CLOSE CONNECTION
>    
>    NEW CONNECTION!!
>    CLOSE CONNECTION
>    [INFO] |yarp.os.impl.PortCoreInputUnit| Removing input from /log/iiticublap199/WalkingModule-0.[walkingModule]/85560 to /yarpjs/proc-walkingmodule-yarptextlogging:i
>    [INFO] |yarp.os.impl.PortCoreInputUnit| Receiving input from /icubSim/left_foot_heel_tiptoe/measures:o to /yarpjs/left_foot_heel_tiptoe/FT:i using tcp
>    /Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:404
>        this.history[id].push(telemetrySample); // update history
>                         ^
>    
>    TypeError: Cannot read property 'push' of undefined
>        at ICubTelemetry.generateTelemetry (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:404:22)
>        at ICubTelemetry.<anonymous> (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:183:18)
>        at Array.forEach (<anonymous>)
>        at ICubTelemetry.<anonymous> (/Users/nunoguedelha/dev/yarp-openmct/iCubTelemVizServer/icubtelemetry.js:182:33)
>        at listOnTimeout (internal/timers.js:555:17)
>        at processTimers (internal/timers.js:498:7)
>    npm ERR! code ELIFECYCLE
>    npm ERR! errno 1
>    npm ERR! yarp-openmct@1.0.0 start: `cd iCubTelemVizServer && node ${NODE_DEBUG_OPTION} iCubTelemVizServer.js`
>    npm ERR! Exit status 1
>    npm ERR!
>    npm ERR! Failed at the yarp-openmct@1.0.0 start script.
>    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
>    
>    npm ERR! A complete log of this run can be found in:
>    npm ERR!     /Users/nunoguedelha/.npm/_logs/2022-10-04T08_39_39_709Z-debug.log
>    ```
> 
>    </details>
> - The issue was fixed with a debug branch `debug/test-with-stefano` pushed to the remote repository. Meanwhile, the robot had to go to maintenance.
> - **[Issue 3]:** Later in the day, @S-Dafarra ran another trial with the debug branch. The issue observed earlier was fixed, no crash ever occurred. We could see the expected Yarplogger messages on the visualiser, but there was a glitch though:
>     - As we ran some rpc command (e.g. `calib all 300`) triggering Yarplogger messages we could not see the messages right away.
>     - Later on, we could only see the messages if we clicked on another entry in the left pane, then on the Yarp Text Logging entry back again.
>     - There is no log for this issue.
> 
> ### Issues recap
> - [ ] **[Issue 1]:** Running the telemetry server from the folder ./iCubTelemVizServer with the node command fails
> - [ ] **[Issue 2]:** Telemetry server crashes at startup when parsing `<robot>/left_foot_heel_tiptoe/measures:o` port FT data.
> - [ ] **[Issue 3]:** We can't see the Yarplogger messages right away as they are received on the **yarpjs** input port, unless we click away from the Yarp Text Logging entry and back again.
> 

_Originally posted by @nunoguedelha in https://github.com/ami-iit/component_ANA-Avatar-XPRIZE/issues/760#issuecomment-1266616475_

I guess that there are some issues on the robot laptop side.

Here the ``npm ls`` output

<details>

<p>

```
icub@icub-console-xps:/usr/local/src/robot/yarp-openmct$ npm ls
yarp-openmct@1.0.0 /usr/local/src/robot/yarp-openmct
├── UNMET DEPENDENCY express@latest
├─┬ express-ws@4.0.0
│ └─┬ ws@5.2.3
│   └── async-limiter@1.0.1
├── UNMET DEPENDENCY jquery@latest
├── UNMET DEPENDENCY openmct@git+https://github.com/ami-iit/openmct.git#7e67bdba9ff7d6c4a98fa952ce484b3d0f474584
├─┬ socket.io@1.5.1
│ ├─┬ debug@2.2.0
│ │ └── ms@0.7.1
│ ├─┬ engine.io@1.7.2
│ │ ├─┬ accepts@1.3.3
│ │ │ ├── mime-types@2.1.34 deduped
│ │ │ └── negotiator@0.6.1
│ │ ├── base64id@0.1.0
│ │ ├─┬ debug@2.2.0
│ │ │ └── ms@0.7.1
│ │ ├─┬ engine.io-parser@1.3.1
│ │ │ ├── after@0.8.1
│ │ │ ├── arraybuffer.slice@0.0.6
│ │ │ ├── base64-arraybuffer@0.1.5
│ │ │ ├── blob@0.0.4
│ │ │ ├─┬ has-binary@0.1.6
│ │ │ │ └── isarray@0.0.1 deduped
│ │ │ └── wtf-8@1.0.0
│ │ └─┬ ws@1.1.1
│ │   ├── options@0.0.6
│ │   └── ultron@1.0.2
│ ├─┬ has-binary@0.1.7
│ │ └── isarray@0.0.1
│ ├─┬ socket.io-adapter@0.4.0
│ │ ├─┬ debug@2.2.0
│ │ │ └── ms@0.7.1
│ │ └─┬ socket.io-parser@2.2.2
│ │   ├── benchmark@1.0.0
│ │   ├── component-emitter@1.1.2 deduped
│ │   ├── debug@0.7.4
│ │   ├── isarray@0.0.1 deduped
│ │   └── json3@3.2.6
│ ├─┬ socket.io-client@1.5.1
│ │ ├── backo2@1.0.2
│ │ ├── component-bind@1.0.0 deduped
│ │ ├── component-emitter@1.2.0
│ │ ├─┬ debug@2.2.0
│ │ │ └── ms@0.7.1
│ │ ├─┬ engine.io-client@1.7.2
│ │ │ ├── component-emitter@1.1.2 deduped
│ │ │ ├── component-inherit@0.0.3
│ │ │ ├─┬ debug@2.2.0
│ │ │ │ └── ms@0.7.1
│ │ │ ├── engine.io-parser@1.3.1 deduped
│ │ │ ├── has-cors@1.1.0
│ │ │ ├── indexof@0.0.1 deduped
│ │ │ ├─┬ parsejson@0.0.1
│ │ │ │ └── better-assert@1.0.2 deduped
│ │ │ ├─┬ parseqs@0.0.2
│ │ │ │ └── better-assert@1.0.2 deduped
│ │ │ ├── parseuri@0.0.4 deduped
│ │ │ ├─┬ ws@1.1.1
│ │ │ │ ├── options@0.0.6 deduped
│ │ │ │ └── ultron@1.0.2 deduped
│ │ │ ├── xmlhttprequest-ssl@1.5.1
│ │ │ └── yeast@0.1.2
│ │ ├── has-binary@0.1.7 deduped
│ │ ├── indexof@0.0.1
│ │ ├── object-component@0.0.3
│ │ ├─┬ parseuri@0.0.4
│ │ │ └─┬ better-assert@1.0.2
│ │ │   └── callsite@1.0.0
│ │ ├── socket.io-parser@2.3.1 deduped
│ │ └── to-array@0.1.4
│ └─┬ socket.io-parser@2.3.1
│   ├── component-emitter@1.1.2
│   ├─┬ debug@2.2.0
│   │ └── ms@0.7.1
│   ├── isarray@0.0.1 deduped
│   └── json3@3.3.2
├── UNMET DEPENDENCY socket.io-stream@latest
├── ws@7.5.7
└─┬ UNMET DEPENDENCY YarpJS@git+https://github.com/robotology/yarp.js.git#08c9c2cd6b6957f45ffb58779bc998252d7c5269
  ├─┬ cmake-js@6.3.0
  │ ├─┬ axios@0.21.4
  │ │ └── follow-redirects@1.14.9
  │ ├─┬ debug@4.3.3
  │ │ └── ms@2.1.2
  │ ├─┬ fs-extra@5.0.0
  │ │ ├── graceful-fs@4.2.9
  │ │ ├─┬ jsonfile@4.0.0
  │ │ │ └── graceful-fs@4.2.9 deduped
  │ │ └── universalify@0.1.2
  │ ├── is-iojs@1.1.0
  │ ├── lodash@4.17.21
  │ ├─┬ memory-stream@0.0.3
  │ │ └─┬ readable-stream@1.0.34
  │ │   ├── core-util-is@1.0.3
  │ │   ├── inherits@2.0.4
  │ │   ├── isarray@0.0.1 deduped
  │ │   └── string_decoder@0.10.31
  │ ├─┬ npmlog@1.2.1
  │ │ ├── ansi@0.3.1
  │ │ ├─┬ are-we-there-yet@1.0.6
  │ │ │ ├── delegates@1.0.0
  │ │ │ └─┬ readable-stream@2.3.7
  │ │ │   ├── core-util-is@1.0.3 deduped
  │ │ │   ├── inherits@2.0.4 deduped
  │ │ │   ├── isarray@1.0.0
  │ │ │   ├── process-nextick-args@2.0.1
  │ │ │   ├── safe-buffer@5.1.2
  │ │ │   ├─┬ string_decoder@1.1.1
  │ │ │   │ └── safe-buffer@5.1.2 deduped
  │ │ │   └── util-deprecate@1.0.2 deduped
  │ │ └─┬ gauge@1.2.7
  │ │   ├── ansi@0.3.1 deduped
  │ │   ├── has-unicode@2.0.1
  │ │   ├── lodash.pad@4.5.1
  │ │   ├── lodash.padend@4.6.1
  │ │   └── lodash.padstart@4.6.1
  │ ├─┬ rc@1.2.8
  │ │ ├── deep-extend@0.6.0
  │ │ ├── ini@1.3.8
  │ │ ├── minimist@1.2.5
  │ │ └── strip-json-comments@2.0.1
  │ ├── semver@5.7.1
  │ ├── splitargs@0.0.7
  │ ├─┬ tar@4.4.19
  │ │ ├── chownr@1.1.4
  │ │ ├─┬ fs-minipass@1.2.7
  │ │ │ └── minipass@2.9.0 deduped
  │ │ ├─┬ minipass@2.9.0
  │ │ │ ├── safe-buffer@5.2.1 deduped
  │ │ │ └── yallist@3.1.1 deduped
  │ │ ├─┬ minizlib@1.3.3
  │ │ │ └── minipass@2.9.0 deduped
  │ │ ├─┬ mkdirp@0.5.5
  │ │ │ └── minimist@1.2.5 deduped
  │ │ ├── safe-buffer@5.2.1 deduped
  │ │ └── yallist@3.1.1
  │ ├─┬ unzipper@0.8.14
  │ │ ├── big-integer@1.6.51
  │ │ ├─┬ binary@0.3.0
  │ │ │ ├── buffers@0.1.1
  │ │ │ └─┬ chainsaw@0.1.0
  │ │ │   └── traverse@0.3.9
  │ │ ├── bluebird@3.4.7
  │ │ ├── buffer-indexof-polyfill@1.0.2
  │ │ ├─┬ duplexer2@0.1.4
  │ │ │ └─┬ readable-stream@2.3.7
  │ │ │   ├── core-util-is@1.0.3 deduped
  │ │ │   ├── inherits@2.0.4 deduped
  │ │ │   ├── isarray@1.0.0
  │ │ │   ├── process-nextick-args@2.0.1 deduped
  │ │ │   ├── safe-buffer@5.1.2
  │ │ │   ├─┬ string_decoder@1.1.1
  │ │ │   │ └── safe-buffer@5.1.2 deduped
  │ │ │   └── util-deprecate@1.0.2 deduped
  │ │ ├─┬ fstream@1.0.12
  │ │ │ ├── graceful-fs@4.2.9 deduped
  │ │ │ ├── inherits@2.0.4 deduped
  │ │ │ ├── mkdirp@0.5.5 deduped
  │ │ │ └─┬ rimraf@2.7.1
  │ │ │   └─┬ glob@7.2.0
  │ │ │     ├── fs.realpath@1.0.0
  │ │ │     ├─┬ inflight@1.0.6
  │ │ │     │ ├── once@1.4.0 deduped
  │ │ │     │ └── wrappy@1.0.2
  │ │ │     ├── inherits@2.0.4 deduped
  │ │ │     ├─┬ minimatch@3.1.2
  │ │ │     │ └─┬ brace-expansion@1.1.11
  │ │ │     │   ├── balanced-match@1.0.2
  │ │ │     │   └── concat-map@0.0.1
  │ │ │     ├─┬ once@1.4.0
  │ │ │     │ └── wrappy@1.0.2 deduped
  │ │ │     └── path-is-absolute@1.0.1
  │ │ ├── listenercount@1.0.1
  │ │ ├─┬ readable-stream@2.1.5
  │ │ │ ├── buffer-shims@1.0.0
  │ │ │ ├── core-util-is@1.0.3 deduped
  │ │ │ ├── inherits@2.0.4 deduped
  │ │ │ ├── isarray@1.0.0
  │ │ │ ├── process-nextick-args@1.0.7
  │ │ │ ├── string_decoder@0.10.31 deduped
  │ │ │ └── util-deprecate@1.0.2
  │ │ └── setimmediate@1.0.5
  │ ├── url-join@0.0.1
  │ ├─┬ which@1.3.1
  │ │ └── isexe@2.0.0
  │ └─┬ yargs@3.32.0
  │   ├── camelcase@2.1.1
  │   ├─┬ cliui@3.2.0
  │   │ ├── string-width@1.0.2 deduped
  │   │ ├─┬ strip-ansi@3.0.1
  │   │ │ └── ansi-regex@2.1.1
  │   │ └─┬ wrap-ansi@2.1.0
  │   │   ├── string-width@1.0.2 deduped
  │   │   └── strip-ansi@3.0.1 deduped
  │   ├── decamelize@1.2.0
  │   ├─┬ os-locale@1.4.0
  │   │ └─┬ lcid@1.0.0
  │   │   └── invert-kv@1.0.0
  │   ├─┬ string-width@1.0.2
  │   │ ├── code-point-at@1.1.0
  │   │ ├─┬ is-fullwidth-code-point@1.0.0
  │   │ │ └── number-is-nan@1.0.1
  │   │ └── strip-ansi@3.0.1 deduped
  │   ├── window-size@0.1.4
  │   └── y18n@3.2.2
  ├─┬ express@4.17.3
  │ ├── accepts@1.3.8 deduped
  │ ├── array-flatten@1.1.1
  │ ├─┬ body-parser@1.19.2
  │ │ ├── bytes@3.1.2
  │ │ ├── content-type@1.0.4 deduped
  │ │ ├── debug@2.6.9 deduped
  │ │ ├── depd@1.1.2 deduped
  │ │ ├─┬ http-errors@1.8.1
  │ │ │ ├── depd@1.1.2 deduped
  │ │ │ ├── inherits@2.0.4 deduped
  │ │ │ ├── setprototypeof@1.2.0 deduped
  │ │ │ ├── statuses@1.5.0 deduped
  │ │ │ └── toidentifier@1.0.1
  │ │ ├─┬ iconv-lite@0.4.24
  │ │ │ └── safer-buffer@2.1.2
  │ │ ├── on-finished@2.3.0 deduped
  │ │ ├── qs@6.9.7 deduped
  │ │ ├─┬ raw-body@2.4.3
  │ │ │ ├── bytes@3.1.2 deduped
  │ │ │ ├── http-errors@1.8.1 deduped
  │ │ │ ├── iconv-lite@0.4.24 deduped
  │ │ │ └── unpipe@1.0.0 deduped
  │ │ └── type-is@1.6.18 deduped
  │ ├─┬ content-disposition@0.5.4
  │ │ └── safe-buffer@5.2.1 deduped
  │ ├── content-type@1.0.4
  │ ├── cookie@0.4.2
  │ ├── cookie-signature@1.0.6
  │ ├─┬ debug@2.6.9
  │ │ └── ms@2.0.0
  │ ├── depd@1.1.2
  │ ├── encodeurl@1.0.2
  │ ├── escape-html@1.0.3
  │ ├── etag@1.8.1
  │ ├─┬ finalhandler@1.1.2
  │ │ ├── debug@2.6.9 deduped
  │ │ ├── encodeurl@1.0.2 deduped
  │ │ ├── escape-html@1.0.3 deduped
  │ │ ├── on-finished@2.3.0 deduped
  │ │ ├── parseurl@1.3.3 deduped
  │ │ ├── statuses@1.5.0 deduped
  │ │ └── unpipe@1.0.0
  │ ├── fresh@0.5.2
  │ ├── merge-descriptors@1.0.1
  │ ├── methods@1.1.2
  │ ├─┬ on-finished@2.3.0
  │ │ └── ee-first@1.1.1
  │ ├── parseurl@1.3.3
  │ ├── path-to-regexp@0.1.7
  │ ├─┬ proxy-addr@2.0.7
  │ │ ├── forwarded@0.2.0
  │ │ └── ipaddr.js@1.9.1
  │ ├── qs@6.9.7
  │ ├── range-parser@1.2.1
  │ ├── safe-buffer@5.2.1
  │ ├─┬ send@0.17.2
  │ │ ├── debug@2.6.9 deduped
  │ │ ├── depd@1.1.2 deduped
  │ │ ├── destroy@1.0.4
  │ │ ├── encodeurl@1.0.2 deduped
  │ │ ├── escape-html@1.0.3 deduped
  │ │ ├── etag@1.8.1 deduped
  │ │ ├── fresh@0.5.2 deduped
  │ │ ├── http-errors@1.8.1 deduped
  │ │ ├── mime@1.6.0
  │ │ ├── ms@2.1.3
  │ │ ├── on-finished@2.3.0 deduped
  │ │ ├── range-parser@1.2.1 deduped
  │ │ └── statuses@1.5.0 deduped
  │ ├─┬ serve-static@1.14.2
  │ │ ├── encodeurl@1.0.2 deduped
  │ │ ├── escape-html@1.0.3 deduped
  │ │ ├── parseurl@1.3.3 deduped
  │ │ └── send@0.17.2 deduped
  │ ├── setprototypeof@1.2.0
  │ ├── statuses@1.5.0
  │ ├─┬ type-is@1.6.18
  │ │ ├── media-typer@0.3.0
  │ │ └── mime-types@2.1.34 deduped
  │ ├── utils-merge@1.0.1
  │ └── vary@1.1.2
  ├── jquery@3.6.0
  ├── nan@2.15.0
  ├─┬ UNMET DEPENDENCY socket.io@4.4.1
  │ ├─┬ accepts@1.3.8
  │ │ ├─┬ mime-types@2.1.34
  │ │ │ └── mime-db@1.51.0
  │ │ └── negotiator@0.6.3
  │ ├── UNMET DEPENDENCY base64id@2.0.0
  │ ├─┬ UNMET DEPENDENCY debug@4.3.3
  │ │ └── UNMET DEPENDENCY ms@2.1.2
  │ ├─┬ UNMET DEPENDENCY engine.io@6.1.3
  │ │ ├── @types/cookie@0.4.1
  │ │ ├── @types/cors@2.8.12
  │ │ ├── @types/node@17.0.21
  │ │ ├── accepts@1.3.8 deduped
  │ │ ├── UNMET DEPENDENCY base64id@2.0.0
  │ │ ├── cookie@0.4.2 deduped
  │ │ ├─┬ cors@2.8.5
  │ │ │ ├── object-assign@4.1.1
  │ │ │ └── vary@1.1.2 deduped
  │ │ ├── UNMET DEPENDENCY debug@4.3.3
  │ │ ├─┬ UNMET DEPENDENCY engine.io-parser@5.0.3
  │ │ │ └── @socket.io/base64-arraybuffer@1.0.2
  │ │ └── UNMET DEPENDENCY ws@8.2.3
  │ ├── UNMET DEPENDENCY socket.io-adapter@2.3.3
  │ └─┬ UNMET DEPENDENCY socket.io-parser@4.0.4
  │   ├── @types/component-emitter@1.2.11
  │   ├── UNMET DEPENDENCY component-emitter@1.3.0
  │   └── UNMET DEPENDENCY debug@4.3.3
  ├─┬ socket.io-stream@0.9.1
  │ ├── component-bind@1.0.0
  │ └─┬ debug@2.2.0
  │   └── ms@0.7.1
  └── three@0.138.0

npm ERR! missing: openmct@git+https://github.com/ami-iit/openmct.git#fix/imagery-flickering, required by yarp-openmct@1.0.0
npm ERR! missing: YarpJS@git+https://github.com/robotology/yarp.js.git, required by yarp-openmct@1.0.0
npm ERR! missing: express@latest, required by yarp-openmct@1.0.0
npm ERR! missing: jquery@latest, required by yarp-openmct@1.0.0
npm ERR! missing: socket.io-stream@latest, required by yarp-openmct@1.0.0
npm ERR! extraneous: openmct@git+https://github.com/ami-iit/openmct.git#7e67bdba9ff7d6c4a98fa952ce484b3d0f474584 /usr/local/src/robot/yarp-openmct/node_modules/openmct
npm ERR! missing: socket.io@4.4.1, required by YarpJS@git+https://github.com/robotology/yarp.js.git#08c9c2cd6b6957f45ffb58779bc998252d7c5269
npm ERR! missing: base64id@2.0.0, required by socket.io@4.4.1
npm ERR! missing: debug@4.3.3, required by socket.io@4.4.1
npm ERR! missing: engine.io@6.1.3, required by socket.io@4.4.1
npm ERR! missing: socket.io-adapter@2.3.3, required by socket.io@4.4.1
npm ERR! missing: socket.io-parser@4.0.4, required by socket.io@4.4.1
npm ERR! missing: ms@2.1.2, required by debug@4.3.3
npm ERR! missing: base64id@2.0.0, required by engine.io@6.1.3
npm ERR! missing: debug@4.3.3, required by engine.io@6.1.3
npm ERR! missing: engine.io-parser@5.0.3, required by engine.io@6.1.3
npm ERR! missing: ws@8.2.3, required by engine.io@6.1.3
npm ERR! missing: component-emitter@1.3.0, required by socket.io-parser@4.0.4
npm ERR! missing: debug@4.3.3, required by socket.io-parser@4.0.4
```

</p>

</details>

How do I fix the unmet dependencies?

--- a/common/openMctServerHandlerBase.js
+++ b/common/openMctServerHandlerBase.js
@@ -12,9 +12,14 @@ OpenMctServerHandlerBase.prototype.CLOSING = 1;
 class Child2ParentCommands {
     static RefreshRegexpConnections = 0;
 }
+class Parent2ChildReplies {
+    static RefreshRegexpConnectionsCompleted = 0;
+    static RefreshRegexpConnectionsFailed = 1;
+}
 Object.freeze(Child2ParentCommands);
 
 module.exports = {
     OpenMctServerHandlerBase,
-    Child2ParentCommands
+    Child2ParentCommands,
+    Parent2ChildReplies
 };

--- a/iCubTelemVizServer/iCubTelemVizServer.js
+++ b/iCubTelemVizServer/iCubTelemVizServer.js
@@ -196,6 +196,7 @@ openMctServerHandler.installRefreshPorts(() => {
     TerminationHandler.prototype.unlistenToNetworkPorts.forEach((disconnect) => {disconnect();});
     TerminationHandler.prototype.unlistenToNetworkPorts = [];
     configHandler.matchRegexpYarpPortNames().then(connectPortsAndStartNotifier);
+    openMctServerHandler.messageChildProcess({rply: 'go ahead!'});
 });
 var ret = openMctServerHandler.start();
 console.log(ret);

--- a/iCubTelemVizServer/iCubTelemVizServer.js
+++ b/iCubTelemVizServer/iCubTelemVizServer.js
@@ -191,12 +191,13 @@ const consoleServerTracker = new WebsocketTracker(consoleServer);
 
 // Create and start the OpenMCT server
 var OpenMctServerHandler = require('./openMctServerHandlerParentProc');
+const {Child2ParentCommands,Parent2ChildReplies} = require("../common/openMctServerHandlerBase");
 var openMctServerHandler = new OpenMctServerHandler(console.log,console.error);
 openMctServerHandler.installRefreshPorts(() => {
     TerminationHandler.prototype.unlistenToNetworkPorts.forEach((disconnect) => {disconnect();});
     TerminationHandler.prototype.unlistenToNetworkPorts = [];
     configHandler.matchRegexpYarpPortNames().then(connectPortsAndStartNotifier);
-    openMctServerHandler.messageChildProcess({rply: 'go ahead!'});
+    openMctServerHandler.messageChildProcess({rply: Parent2ChildReplies.RefreshRegexpConnectionsCompleted});
 });
 var ret = openMctServerHandler.start();
 console.log(ret);

--- a/iCubTelemVizServer/openMctServerHandlerParentProc.js
+++ b/iCubTelemVizServer/openMctServerHandlerParentProc.js
@@ -25,6 +25,7 @@ function OpenMctServerHandlerParentProc(outputCallback,errorCallback) {
         this.errorCallback('Untriggered closure: ' + errorObject.toString());
     };
     this.refreshPortsNconnections = () => {};
+    this.npmStart = undefined;
 }
 
 // Old Javascript inheritance: inherit all of OpenMctServerHandlerBase'methods
@@ -47,23 +48,23 @@ OpenMctServerHandlerParentProc.prototype.start = function () {
 
     // Start the process
     const wPath = path.join(process.cwd(), '..', 'openmctStaticServer');
-    const npmStart = childProcess.spawn('sh', ['runModule.sh'], {
+    this.npmStart = childProcess.spawn('sh', ['runModule.sh'], {
         shell: 'bash',
         cwd: wPath,
         stdio: ['pipe', 'pipe', 'pipe', 'ipc']
     });
 
     // Set the output callbacks
-    npmStart.stdout.on('data', function (data) {
+    this.npmStart.stdout.on('data', function (data) {
         embeddedThis.outputCallback('[OPEN-MCT STATIC SERVER] stdout: ' + data);
     });
-    npmStart.stderr.on('data', function (data) {
+    this.npmStart.stderr.on('data', function (data) {
         embeddedThis.outputCallback('[OPEN-MCT STATIC SERVER] stderr: ' + data);
     });
-    npmStart.on('error', function (error) {
+    this.npmStart.on('error', function (error) {
         embeddedThis.outputCallback('[OPEN-MCT STATIC SERVER] error: ' + error.message);
     });
-    npmStart.on('message', function (m) {
+    this.npmStart.on('message', function (m) {
         embeddedThis.outputCallback('[OPEN-MCT STATIC SERVER] ipc: ' + JSON.stringify(m));
         Object.keys(m).forEach((k) => {
             switch (k) {
@@ -83,7 +84,7 @@ OpenMctServerHandlerParentProc.prototype.start = function () {
             }
         });
     });
-    npmStart.on('close', function (code,signal) {
+    this.npmStart.on('close', function (code,signal) {
         embeddedThis.processPID = undefined;
         // Call local or synchronisation callback
         try {
@@ -123,15 +124,26 @@ OpenMctServerHandlerParentProc.prototype.stop = function (signal,onCloseSuccess,
         this.onCloseSuccess = onCloseSuccess;
         this.onCloseFailure = onCloseFailure;
         process.kill(this.processPID,signal);
+        this.npmStart = undefined;
         return {status: 'WRPLY', message: util.format('Process (PID %d) OpenMCT Server stopping (signal %s) ...',this.processPID,signal)};
     } else {
         onCloseSuccess('');
+        this.npmStart = undefined;
         return {status: 'WARNING', message: 'No process OpenMCT Server running...'};
     }
 }
 
 OpenMctServerHandlerParentProc.prototype.isOn = function () {
     return (this.processPID !== undefined);
+}
+
+// Method for sending messages to the child process through IPC only if there is one
+OpenMctServerHandlerParentProc.prototype.messageChildProcess = function (message) {
+    if (!this.npmStart.connected) {
+        console.warn(`Could not send message. Connection lost with the child process!`);
+        return false;
+    }
+    return this.npmStart.send(message);
 }
 
 module.exports = OpenMctServerHandlerParentProc;

--- a/openmctStaticServer/openMctServerHandlerChildProc.js
+++ b/openmctStaticServer/openMctServerHandlerChildProc.js
@@ -1,7 +1,7 @@
 "use strict";
 
 // Import comon properties for OpenMctServerHandlerParentProc and OpenMctServerHandlerChildProc
-const {OpenMctServerHandlerBase,Child2ParentCommands} = require('../common/openMctServerHandlerBase');
+const {OpenMctServerHandlerBase,Child2ParentCommands,Parent2ChildReplies} = require('../common/openMctServerHandlerBase');
 
 function OpenMctServerHandlerChildProc(outputCallback) {
     // Old Javascript inheritance: Apply the "parent" class OpenMctServerHandlerBase
@@ -32,13 +32,26 @@ OpenMctServerHandlerChildProc.prototype.requestPortsRefresh = function () {
     return new Promise(function (resolve, reject) {
         process.once('message', function (m) {
             console.log(`[ICUB-TELEM-VIZ SERVER] message: ${JSON.stringify(m)}`);
-            resolve('Ports refreshing completed!');
+            if (!("rply" in m)) {
+                reject(`Can't parse IPC reply`);
+            } else {
+                switch (m.rply) {
+                    case Parent2ChildReplies.RefreshRegexpConnectionsCompleted:
+                        resolve('Ports refreshing completed!');
+                        break;
+                    case Parent2ChildReplies.RefreshRegexpConnectionsFailed:
+                        reject(`Ports refreshing Failed`);
+                        break;
+                    default:
+                }
+            };
         });
         if (!this.messageParentProcess({"cmd": Child2ParentCommands.RefreshRegexpConnections})) {
             reject('Could not send request for refreshing ports!');
-        };
-        console.log('Request for refreshing ports sent successfully');
-    });
+        } else {
+            console.log('Request for refreshing ports sent successfully');
+        }
+    }.bind(this));
 }
 
 module.exports = OpenMctServerHandlerChildProc;

--- a/openmctStaticServer/openMctServerHandlerChildProc.js
+++ b/openmctStaticServer/openMctServerHandlerChildProc.js
@@ -29,7 +29,10 @@ OpenMctServerHandlerChildProc.prototype.reportPIDtoParent = function () {
 }
 
 OpenMctServerHandlerChildProc.prototype.requestPortsRefresh = function () {
-    return this.messageParentProcess({"cmd": Child2ParentCommands.RefreshRegexpConnections});
+    if (!this.messageParentProcess({"cmd": Child2ParentCommands.RefreshRegexpConnections})) {
+        return Promise.reject('Could not send request for refreshing ports. Connection lost with the parent process!');
+    };
+    return Promise.resolve('Request for refreshing ports sent successfully');
 }
 
 module.exports = OpenMctServerHandlerChildProc;

--- a/openmctStaticServer/openMctServerHandlerChildProc.js
+++ b/openmctStaticServer/openMctServerHandlerChildProc.js
@@ -17,7 +17,7 @@ OpenMctServerHandlerChildProc.prototype.constructor = OpenMctServerHandlerChildP
 // Method for sending messages to the parent process through IPC only if there is one
 OpenMctServerHandlerChildProc.prototype.messageParentProcess = function (message) {
     if (!process.connected) {
-        console.warn(`Could not send request for refreshing ports. Connection lost with the parent process!`);
+        console.warn(`Could not send message. Connection lost with the parent process!`);
         return false;
     }
     return process.send(message);
@@ -29,10 +29,16 @@ OpenMctServerHandlerChildProc.prototype.reportPIDtoParent = function () {
 }
 
 OpenMctServerHandlerChildProc.prototype.requestPortsRefresh = function () {
-    if (!this.messageParentProcess({"cmd": Child2ParentCommands.RefreshRegexpConnections})) {
-        return Promise.reject('Could not send request for refreshing ports. Connection lost with the parent process!');
-    };
-    return Promise.resolve('Request for refreshing ports sent successfully');
+    return new Promise(function (resolve, reject) {
+        process.once('message', function (m) {
+            console.log(`[ICUB-TELEM-VIZ SERVER] message: ${JSON.stringify(m)}`);
+            resolve('Ports refreshing completed!');
+        });
+        if (!this.messageParentProcess({"cmd": Child2ParentCommands.RefreshRegexpConnections})) {
+            reject('Could not send request for refreshing ports!');
+        };
+        console.log('Request for refreshing ports sent successfully');
+    });
 }
 
 module.exports = OpenMctServerHandlerChildProc;

--- a/openmctStaticServer/server.js
+++ b/openmctStaticServer/server.js
@@ -34,10 +34,12 @@ const staticServer = new StaticServer();
 // The default page request processing is defined by default, but redefined here for triggering a refresh of the port
 // connections.
 app.get('/', function(req, res){
-    res.sendFile('index.html',{ root : __dirname});
-    if (!procHandler.requestPortsRefresh()) {
-        console.warn('Ports refresh request not sent!');
-    }
+    procHandler.requestPortsRefresh().then((result) => {
+        console.log(result);
+        res.sendFile('index.html',{ root : __dirname});
+    }).catch((errorMessage) => {
+        console.error(errorMessage);
+    });
 });
 
 app.get('/config/confServers.json', function(req, res){

